### PR TITLE
Add direct access token auth

### PR DESCRIPTION
**Pull request by ionutcalara**

### CODE REVIEW

File: src/Credentials.php [hunk 1]
```diff
@@ -89,6 +94,12 @@ private static function validateCredentials($credentials) {
             return $credentials;
         }
 
+        private function maybeInitializeAccessToken(){
```
Comment (ChrisSimmons) [inline src/Credentials.php]: Can you change the name of this?  `initializeAccessToken` would be fine.

**Review by ChrisSimmons — APPROVED** [review]
Except for the one method name change, I'm good with this.  Initially I was hesitant on the change as the recommended pattern is to use a singleton client.  However, with resource owner this can admittedly become cumbersome where it would be easier to simply persist the access token and "rehydrate" a client for a given user.

File: src/Credentials.php [hunk 2]
```diff
@@ -75,6 +76,10 @@ private static function validateCredentials($credentials) {
                 $credentials["credential_type"] = "resource_owner";
             }
 
+            if(array_key_exists("access_token",$credentials)) {
+                $credentials["credential_type"] = "resource_owner";
```
Comment (mapitman) [inline src/Credentials.php]: An access token being present does not necessarily mean the grant type is resource owner.

**Review by mapitman — CHANGES_REQUESTED** [review]

File: src/Credentials.php (hunk 2 above)
Comment (ChrisSimmons) [reply src/Credentials.php]: Good catch.

Comment (ChrisSimmons) [reply src/Credentials.php]: If I read this correctly, this if block can be removed completely, deferring to the check that comes immediately afterward (checking for the presence of `username` and `password`).

File: src/Credentials.php (hunk 1 above)
Comment (ionutcalara) [reply src/Credentials.php]: I have updated the method name as requested.

File: src/Credentials.php (hunk 2 above)
Comment (ionutcalara) [reply src/Credentials.php]: I have removed the additional check, as access token won't work without something else like refresh token.

**Comment by ChrisSimmons** [comment]
My apologies for the delay on this.  Merging now.

**Pull request merged**